middle_end/flambda/simplify/simplify_expr.rec.ml

```diff
@@ -691,29 +691,48 @@ Format.eprintf "Simplifying inlined body with DE depth delta = %d\n%!"
   *)
     simplify_expr dacc inlined k
   | None ->
-    let dacc, use_id =
-      DA.record_continuation_use dacc (Apply.continuation apply) Non_inlinable
-        ~typing_env_at_use:(DA.typing_env dacc)
-        ~arg_types:(T.unknown_types_from_arity result_arity)
-    in
-    let dacc, exn_cont_use_id =
-      DA.record_continuation_use dacc
-        (Exn_continuation.exn_handler (Apply.exn_continuation apply))
-        Non_inlinable
-        ~typing_env_at_use:(DA.typing_env dacc)
-        ~arg_types:(T.unknown_types_from_arity (
-          Exn_continuation.arity (Apply.exn_continuation apply)))
-    in
-    let user_data, uacc = k dacc in
-    let apply =
-      Simplify_common.update_exn_continuation_extra_args uacc ~exn_cont_use_id
-        apply
-    in
-    let expr =
-      Simplify_common.add_wrapper_for_fixed_arity_apply uacc ~use_id
-        result_arity apply
-    in
-    expr, user_data, uacc
+    begin match Apply.continuation apply with
```

Comment:
There's a lot of duplicated code, it would be nice if you could factor out the common parts

Reply:
(Besides, I don't think we generate non-returning direct calls at the moment)

Reply:
I factorized the code properly

Reply:
I forgot to leave a comment about it, but there's some factoring possible for the C call case too.